### ISSUE TITLE
NOT UPDATIN BY IT SELF [BUG] 

### ISSUE THREAD

**Issue by FamSkovGreve** [kpoppel/homeassistant-eforsyning#36]
Hey. 

There is a problem, with the updating in this plugin. 

You should be your selv restart the Plugin if you want new data from the Eforsyning. 

Can you fix this ?

**Comment by kpoppel** [comment]
Please give some more information, log entries, debug information you turned on in `configuration.yaml` etc. 
 I cannot reproduce this on my end without more information, as I see data from every 24 hours for the last 30 days on my end.

**Comment by jonas-jespersen** [comment]
I have had the same problem for about a month now. When I restart Home Assistant all data from eForsyning is updated as expected to actual data, but then it doesnt updata again until next restart.

This is what I could find in the log regarding the eForsyning plugin:

2022-09-09 06:26:19.991 ERROR (MainThread) [custom_components.eforsyning.coordinator] Unexpected error fetching eForsyning data: Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/requests/models.py", line 971, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/simplejson/__init__.py", line 525, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.10/site-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/local/lib/python3.10/site-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/eforsyning/coordinator.py", line 53, in _async_update_data
    data = await self.hass.async_add_executor_job(self.api.get_latest)
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/eforsyning/pyeforsyning/eforsyning.py", line 347, in get_latest
    self._get_intallations()
  File "/config/custom_components/eforsyning/pyeforsyning/eforsyning.py", line 80, in _get_intallations
    ebrugerid = self._get_ebrugerid()
  File "/config/custom_components/eforsyning/pyeforsyning/eforsyning.py", line 58, in _get_ebrugerid
    result_json = result.json()
  File "/usr/local/lib/python3.10/site-packages/requests/models.py", line 975, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 205, in _async_refresh
    self.data = await self._async_update_data()
  File "/config/custom_components/eforsyning/coordinator.py", line 55, in _async_update_data
    raise ConfigEntryNotReady from error

**Comment by kpoppel** [comment]
Okay @jonas-jespersen that is new. Could you try to enable debug logging and provide the log on this, please. Refer to the README for instruction. It almost seem like you did not succeed the login with an error like this. If I can figure out what causes this at least I can make the integration explain what is is wrong a little better.

**Comment by jonas-jespersen** [comment]
@kpoppel I have already set up:

logger: 
  default: info
  logs: 
    pyeforsyning.eforsyning: debug

as described in the RADME in my configuration.yaml, and the above is all could find in the home-assistant.log file regarding eForsyning. 

I'm not sure what else is needed?

**Comment by kpoppel** [comment]
Nothing more, except I just noticed the README instruction is wrong (now fixed).
Instead of `pyeforsyning.eforsyning: debug` you need to put `custom_components.eforsyning: debug`. Then a lot more information should be dumped in the log, hopefully revealing a little more.  In the code right around this error (eforsyning.py line 58) you could copy the `_LOGGER.debug()` line below to print `result.text` before that line.

I suspect though that something unusual happened before that time during the login process.

**Comment by jonas-jespersen** [comment]
This is what I got in the log so far. Timestamp at 2022-09-11 13:00 was where I restarted Home Assistant last time, and at this time the data was updated. However, since then data was not updated.

2022-09-11 13:00:21.385 INFO (MainThread) [homeassistant.setup] Setting up eforsyning
2022-09-11 13:00:21.385 INFO (MainThread) [homeassistant.setup] Setup of domain eforsyning took 0.0 seconds
2022-09-11 13:00:21.498 DEBUG (MainThread) [custom_components.eforsyning] eForsyning ConfigData: {'username': 'XXXXXXXXX', 'password': 'XXXX', 'supplierid': 'XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX', 'entityname': 'EForsyning', 'billing_period_skew': False, 'is_water_supply': False}
2022-09-11 13:00:21.499 DEBUG (SyncWorker_4) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting api server at supplier XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-11 13:00:21.677 DEBUG (SyncWorker_4) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting api server https://api2.dff-edb.dk/XXXXXX/
2022-09-11 13:00:21.677 DEBUG (SyncWorker_4) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting access token
2022-09-11 13:00:21.819 DEBUG (SyncWorker_4) [custom_components.eforsyning.pyeforsyning.eforsyning] Got access token: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-11 13:00:22.041 DEBUG (SyncWorker_4) [custom_components.eforsyning.pyeforsyning.eforsyning] Login success
2022-09-11 13:00:22.139 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting latest data
2022-09-11 13:00:22.139 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting installations at supplier: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-11 13:00:22.139 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting userinfo from API (ebrugerinfo)
2022-09-11 13:00:22.139 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Trying: https://api2.dff-edb.dk/XXXXXX/api/getebrugerinfo?id=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-11 13:00:22.337 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}
2022-09-11 13:00:22.337 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}
2022-09-11 13:00:22.337 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Response from userinfo API. ebrugerinfo: 200, Body: {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}, ebruger: XXXXX
2022-09-11 13:00:22.337 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Trying: https://api2.dff-edb.dk/XXXXXX/api/FindInstallationer?id=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-11 13:00:22.625 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Response from API. Status: 200, Body: {"Installationer":[{"EjendomNr":XXXXXXXXX,"Adresse":"XXXXXXXX XX","InstallationNr":1,"ForbrugerNr":"XXXXXXXXX","MålerNr":"XXXXXXXX","By":"XXXXXX","PostNr":"XXXX","AktivNr":2,"Målertype":"XXXXXXXX"}]}
2022-09-11 13:00:22.625 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting installatons[0] {'EjendomNr': XXXXXXXXX, 'Adresse': 'XXXXXXXX XX', 'InstallationNr': 1, 'ForbrugerNr': 'XXXXXXXXX', 'MålerNr': 'XXXXXXXX', 'By': 'XXXXXX', 'PostNr': 'XXXX', 'AktivNr': 2, 'Målertype': 'XXXXXXXX'}
2022-09-11 13:00:22.625 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting time series
2022-09-11 13:00:22.625 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] POST data to API. {'Ejendomnr': 'XXXXXXXXX', 'AktivNr': '2', 'I_Nr': '1', 'AarsMaerke': 2022, 'ForbrugsAfgraensning_FraDato': '10-09-2022', 'ForbrugsAfgraensning_TilDato': '11-09-2022', 'ForbrugsAfgraensning_FraAflaesning': '0', 'ForbrugsAfgraensning_TilAflaesning': '2', 'ForbrugsAfgraensning_MedtagMellemliggendeMellemaflas': 'true', 'Optioner': 'foBestemtBeboer, foSkabDetaljer, foMedtagWebAflaes', 'AHoejDetail': 'false', 'Aflaesningsfilter': 'afDagsvis', 'AflaesningsFilterDag': 'ULTIMO', 'AflaesningsUdjaevning': 'true', 'SletFiltreredeAflaesninger': 'true', 'MedForventetForbrug': 'true', 'OmregnForbrugTilAktuelleEnhed': 'true'}
2022-09-11 13:00:24.409 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting time series 200, Body: {
2022-09-11 13:00:24.420 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Parsing results - heating metering
2022-09-11 13:00:24.459 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing results
2022-09-11 13:00:24.459 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting billing details at supplier XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-11 13:00:24.459 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] POST to API
2022-09-11 13:00:25.547 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting billing details 200
2022-09-11 13:00:25.548 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] {
2022-09-11 13:00:25.548 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Parsing results - billing
2022-09-11 13:00:25.548 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing results
2022-09-11 13:00:25.548 DEBUG (SyncWorker_2) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing latest data
2022-09-11 13:00:25.665 DEBUG (MainThread) [custom_components.eforsyning.coordinator] Finished fetching eForsyning data in 4.167 seconds (success: True)
2022-09-11 13:00:28.750 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.eforsyning
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Water Temperature forward
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Water Temperature return
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Water Temperature exp-return
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Water Temperature cooling
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Energy start
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Energy end
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Energy used
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Energy exp-used
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Energy exp-end
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Energy total-used
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Energy use-prognosis
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Water start
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Water end
2022-09-11 13:00:28.779 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Water used
2022-09-11 13:00:28.780 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Water exp-used
2022-09-11 13:00:28.780 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Water exp-end
2022-09-11 13:00:28.780 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Water total-used
2022-09-11 13:00:28.780 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Water use-prognosis
2022-09-11 13:00:28.780 DEBUG (MainThread) [custom_components.eforsyning.sensor] Registering Sensor for Amount remaining
2022-09-11 19:00:28.001 DEBUG (SyncWorker_0) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting api server at supplier XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-11 19:00:28.219 DEBUG (SyncWorker_0) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting api server https://api2.dff-edb.dk/XXXXXX/
2022-09-11 19:00:28.219 DEBUG (SyncWorker_0) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting access token
2022-09-11 19:00:28.423 DEBUG (SyncWorker_0) [custom_components.eforsyning.pyeforsyning.eforsyning] Got access token: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-11 19:00:28.681 DEBUG (SyncWorker_0) [custom_components.eforsyning.pyeforsyning.eforsyning] Login success
2022-09-11 19:00:28.681 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting latest data
2022-09-11 19:00:28.682 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting installations at supplier: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-11 19:00:28.682 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting userinfo from API (ebrugerinfo)
2022-09-11 19:00:28.682 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Trying: https://api2.dff-edb.dk/XXXXXX/api/getebrugerinfo?id=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-11 19:00:28.860 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}
2022-09-11 19:00:28.860 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}
2022-09-11 19:00:28.860 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Response from userinfo API. ebrugerinfo: 200, Body: {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}, ebruger: XXXXX
2022-09-11 19:00:28.860 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Trying: https://api2.dff-edb.dk/XXXXXX/api/FindInstallationer?id=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-11 19:00:29.157 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Response from API. Status: 200, Body: {"Installationer":[{"EjendomNr":XXXXXXXXX,"Adresse":"XXXXXXXX XX","InstallationNr":1,"ForbrugerNr":"XXXXXXXXX","MålerNr":"XXXXXXXX","By":"XXXXXX","PostNr":"XXXX","AktivNr":2,"Målertype":"XXXXXXXX"}]}
2022-09-11 19:00:29.157 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting installatons[0] {'EjendomNr': XXXXXXXXX, 'Adresse': 'XXXXXXXX XX', 'InstallationNr': 1, 'ForbrugerNr': 'XXXXXXXXX', 'MålerNr': 'XXXXXXXX', 'By': 'XXXXXX', 'PostNr': 'XXXX', 'AktivNr': 2, 'Målertype': 'XXXXXXXX'}
2022-09-11 19:00:29.157 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting time series
2022-09-11 19:00:29.158 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] POST data to API. {'Ejendomnr': 'XXXXXXXXX', 'AktivNr': '2', 'I_Nr': '1', 'AarsMaerke': 2022, 'ForbrugsAfgraensning_FraDato': '10-09-2022', 'ForbrugsAfgraensning_TilDato': '11-09-2022', 'ForbrugsAfgraensning_FraAflaesning': '0', 'ForbrugsAfgraensning_TilAflaesning': '2', 'ForbrugsAfgraensning_MedtagMellemliggendeMellemaflas': 'true', 'Optioner': 'foBestemtBeboer, foSkabDetaljer, foMedtagWebAflaes', 'AHoejDetail': 'false', 'Aflaesningsfilter': 'afDagsvis', 'AflaesningsFilterDag': 'ULTIMO', 'AflaesningsUdjaevning': 'true', 'SletFiltreredeAflaesninger': 'true', 'MedForventetForbrug': 'true', 'OmregnForbrugTilAktuelleEnhed': 'true'}
2022-09-11 19:00:30.672 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting time series 200, Body: {
2022-09-11 19:00:30.704 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Parsing results - heating metering
2022-09-11 19:00:30.815 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing results
2022-09-11 19:00:30.815 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting billing details at supplier XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-11 19:00:30.815 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] POST to API
2022-09-11 19:00:31.919 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting billing details 200
2022-09-11 19:00:31.919 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] {
2022-09-11 19:00:31.920 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Parsing results - billing
2022-09-11 19:00:31.920 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing results
2022-09-11 19:00:31.920 DEBUG (SyncWorker_7) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing latest data
2022-09-11 19:00:31.921 DEBUG (MainThread) [custom_components.eforsyning.coordinator] Finished fetching eForsyning data in 3.920 seconds (success: True)
2022-09-12 01:00:31.004 DEBUG (SyncWorker_9) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting api server at supplier XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-12 01:00:31.235 DEBUG (SyncWorker_9) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting api server https://api2.dff-edb.dk/XXXXXX/
2022-09-12 01:00:31.235 DEBUG (SyncWorker_9) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting access token
2022-09-12 01:00:33.288 DEBUG (SyncWorker_9) [custom_components.eforsyning.pyeforsyning.eforsyning] Got access token: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-12 01:00:33.463 DEBUG (SyncWorker_9) [custom_components.eforsyning.pyeforsyning.eforsyning] Login success
2022-09-12 01:00:33.464 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting latest data
2022-09-12 01:00:33.464 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting installations at supplier: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-12 01:00:33.464 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting userinfo from API (ebrugerinfo)
2022-09-12 01:00:33.464 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Trying: https://api2.dff-edb.dk/XXXXXX/api/getebrugerinfo?id=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-12 01:00:33.689 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}
2022-09-12 01:00:33.689 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}
2022-09-12 01:00:33.689 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Response from userinfo API. ebrugerinfo: 200, Body: {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}, ebruger: XXXXX
2022-09-12 01:00:33.689 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Trying: https://api2.dff-edb.dk/XXXXXX/api/FindInstallationer?id=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-12 01:00:34.176 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Response from API. Status: 200, Body: {"Installationer":[{"EjendomNr":XXXXXXXXX,"Adresse":"XXXXXXXX XX","InstallationNr":1,"ForbrugerNr":"XXXXXXXXX","MålerNr":"XXXXXXXX","By":"XXXXXX","PostNr":"XXXX","AktivNr":2,"Målertype":"XXXXXXXX"}]}
2022-09-12 01:00:34.177 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting installatons[0] {'EjendomNr': XXXXXXXXX, 'Adresse': 'XXXXXXXX XX', 'InstallationNr': 1, 'ForbrugerNr': 'XXXXXXXXX', 'MålerNr': 'XXXXXXXX', 'By': 'XXXXXX', 'PostNr': 'XXXX', 'AktivNr': 2, 'Målertype': 'XXXXXXXX'}
2022-09-12 01:00:34.177 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting time series
2022-09-12 01:00:34.177 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] POST data to API. {'Ejendomnr': 'XXXXXXXXX', 'AktivNr': '2', 'I_Nr': '1', 'AarsMaerke': 2022, 'ForbrugsAfgraensning_FraDato': '11-09-2022', 'ForbrugsAfgraensning_TilDato': '12-09-2022', 'ForbrugsAfgraensning_FraAflaesning': '0', 'ForbrugsAfgraensning_TilAflaesning': '2', 'ForbrugsAfgraensning_MedtagMellemliggendeMellemaflas': 'true', 'Optioner': 'foBestemtBeboer, foSkabDetaljer, foMedtagWebAflaes', 'AHoejDetail': 'false', 'Aflaesningsfilter': 'afDagsvis', 'AflaesningsFilterDag': 'ULTIMO', 'AflaesningsUdjaevning': 'true', 'SletFiltreredeAflaesninger': 'true', 'MedForventetForbrug': 'true', 'OmregnForbrugTilAktuelleEnhed': 'true'}
2022-09-12 01:00:35.770 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting time series 200, Body: {
2022-09-12 01:00:35.780 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Parsing results - heating metering
2022-09-12 01:00:35.811 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing results
2022-09-12 01:00:35.812 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting billing details at supplier XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-12 01:00:35.812 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] POST to API
2022-09-12 01:00:37.213 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting billing details 200
2022-09-12 01:00:37.214 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] {
2022-09-12 01:00:37.215 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Parsing results - billing
2022-09-12 01:00:37.215 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing results
2022-09-12 01:00:37.215 DEBUG (SyncWorker_8) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing latest data
2022-09-12 01:00:37.218 DEBUG (MainThread) [custom_components.eforsyning.coordinator] Finished fetching eForsyning data in 6.217 seconds (success: True)
2022-09-12 07:00:37.001 DEBUG (SyncWorker_10) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting api server at supplier XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-12 07:00:37.262 DEBUG (SyncWorker_10) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting api server https://api2.dff-edb.dk/XXXXXX/
2022-09-12 07:00:37.262 DEBUG (SyncWorker_10) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting access token
2022-09-12 07:01:58.867 ERROR (SyncWorker_10) [custom_components.eforsyning.pyeforsyning.eforsyning] Not able to get access token. HTTP status: {result.status_code}. Probably a wrong username.
2022-09-12 07:01:58.877 DEBUG (MainThread) [custom_components.eforsyning.coordinator] Finished

**Comment by kpoppel** [comment]
Thanks for the detailed log. The access token failing is the effect of getting silently blocked by the Api service. It is really a single call to get a token, and getting blocked denies access to that URL.  While I cannot fix how local settings at the Api server does this, you might try to set the const.py MIN_TIME... constant to 12 or 24 hours as a first attempt.
If you use NabuCasa that could be an issue too as those IP addresses may get blocked by default because the sum of eforsyning users make too many accesses to get flagged as unsafe.
 But try to increase the poll time and see how it goes.
 Also if you try to access eforsyning after seeing this error in the integration, try to login using your browser from the same IP. You should notice being blocked there too. Do the same from your mobile phone, and notice that works.

**Comment by jonas-jespersen** [comment]
Ok, I will try to decrease the update interval as suggested, and see what happens.

However, later this afternoon after posting the last log, I got the following entries in the log. It seems that this time login was successful and I got the access token. However, the data did still not update in Home Assistant.

Also if I am actually blocked by the server, how come that I can get the data to update by restarting Home Assistant.

2022-09-12 13:01:58.002 DEBUG (SyncWorker_6) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting api server at supplier XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-12 13:01:58.248 DEBUG (SyncWorker_6) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting api server https://api2.dff-edb.dk/XXXXXX/
2022-09-12 13:01:58.248 DEBUG (SyncWorker_6) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting access token
2022-09-12 13:01:59.894 DEBUG (SyncWorker_6) [custom_components.eforsyning.pyeforsyning.eforsyning] Got access token: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-12 13:02:00.114 DEBUG (SyncWorker_6) [custom_components.eforsyning.pyeforsyning.eforsyning] Login success
2022-09-12 13:02:00.114 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting latest data
2022-09-12 13:02:00.114 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting installations at supplier: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-12 13:02:00.114 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting userinfo from API (ebrugerinfo)
2022-09-12 13:02:00.114 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Trying: https://api2.dff-edb.dk/XXXXXX/api/getebrugerinfo?id=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-12 13:02:00.275 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}
2022-09-12 13:02:00.275 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}
2022-09-12 13:02:00.275 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Response from userinfo API. ebrugerinfo: 200, Body: {"id":XXXXX,"navn":"XXXXXXXX","navn2":"","vej":"XXXXXXXX","husnr":"XX","litra":"","litra2":"","postnr":"XXXX","bynavn":"XXXXXX","email":"","beboeremail":"XXXXXX","tlf":"XXXXXXXX","rolle":1,"indflyttet":"XX-XX-XXXX","notifikationmail":"","notifikationmobil":"","notifikationtype":"","ejendomnr":XXXXXXXXX,"frekvens":0,"overvaagningstypeforbrug":0,"vaerdiforbrug":0,"overvaagningstypeinstallation":0,"vaerdiinstallation":0,"ejerNavn":"XXXXXXXX","menupunkter":["fakturagodkendelse","opgaver","noter","mitforbrug","minedokumenter","sammenlignforbrug","mineoplysninger","beskeder","calculatedvarmeregnskab","meldflytning","smsservice","kontakt","indtastaflaesninger"],"huskelisteId":0}, ebruger: XXXXX
2022-09-12 13:02:00.276 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Trying: https://api2.dff-edb.dk/XXXXXX/api/FindInstallationer?id=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-09-12 13:02:00.646 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Response from API. Status: 200, Body: {"Installationer":[{"EjendomNr":XXXXXXXXX,"Adresse":"XXXXXXXX XX","InstallationNr":1,"ForbrugerNr":"XXXXXXXXX","MålerNr":"XXXXXXXX","By":"XXXXXX","PostNr":"XXXX","AktivNr":2,"Målertype":"XXXXXXXX"}]}
2022-09-12 13:02:00.646 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting installatons[0] {'EjendomNr': XXXXXXXXX, 'Adresse': 'XXXXXXXX XX', 'InstallationNr': 1, 'ForbrugerNr': 'XXXXXXXXX', 'MålerNr': 'XXXXXXXX', 'By': 'XXXXXX', 'PostNr': 'XXXX', 'AktivNr': 2, 'Målertype': 'XXXXXXXX'}
2022-09-12 13:02:00.646 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting time series
2022-09-12 13:02:00.646 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] POST data to API. {'Ejendomnr': 'XXXXXXXXX', 'AktivNr': '2', 'I_Nr': '1', 'AarsMaerke': 2022, 'ForbrugsAfgraensning_FraDato': '11-09-2022', 'ForbrugsAfgraensning_TilDato': '12-09-2022', 'ForbrugsAfgraensning_FraAflaesning': '0', 'ForbrugsAfgraensning_TilAflaesning': '2', 'ForbrugsAfgraensning_MedtagMellemliggendeMellemaflas': 'true', 'Optioner': 'foBestemtBeboer, foSkabDetaljer, foMedtagWebAflaes', 'AHoejDetail': 'false', 'Aflaesningsfilter': 'afDagsvis', 'AflaesningsFilterDag': 'ULTIMO', 'AflaesningsUdjaevning': 'true', 'SletFiltreredeAflaesninger': 'true', 'MedForventetForbrug': 'true', 'OmregnForbrugTilAktuelleEnhed': 'true'}
2022-09-12 13:02:02.295 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting time series 200, Body: {
2022-09-12 13:02:02.343 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Parsing results - heating metering
2022-09-12 13:02:02.438 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing results
2022-09-12 13:02:02.438 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Getting billing details at supplier XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
2022-09-12 13:02:02.438 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] POST to API
2022-09-12 13:02:03.684 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Done getting billing details 200
2022-09-12 13:02:03.685 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] {
2022-09-12 13:02:03.686 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Parsing results - billing
2022-09-12 13:02:03.686 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing results
2022-09-12 13:02:03.686 DEBUG (SyncWorker_5) [custom_components.eforsyning.pyeforsyning.eforsyning] Done parsing latest data
2022-09-12 13:02:03.689 DEBUG (MainThread) [custom_components.eforsyning.coordinator] Finished fetching eForsyning data in 5.688 seconds (success: True)

**Comment by kpoppel** [comment]
That is a very good question.  The log from 13:01 today clearly indicates login and data retrieval was a success, meaning the "non-Home Assistant" specific part works as intended. I assume you see a lot of JOSN data dumped as well you cut away (thanks).  I am by no means an expert in Home Assistant's inner workings, but have modelled the integration after the same principles the core integrations are built on, and how the documentation and code template suggests the HA-part should be built.
  I will keep a lookout here too, but I am not sure how to reproduce this.

**Comment by kpoppel** [comment]
Okay, good news.  I have the solution for the sensors not updating when new data is available.  You logs above hinted a few things for me - and it occurred I have on average restarted my own Home Assistant every 1-2 days, so I haven't ever noticed this effect.
New release coming up to fix this.

As for the access token I cannot do anything about that except lowering the fetch frequency.  The source data are only updated every 24 hours after midnight anyhow.

If you are using NabuCasa, maybe try disabling it and see if this has an effect.  If you try this and it makes a difference, please open a new issue to get the documentation updated with this information to help others experiencing the same thing. (This could be a duck)

**Comment by jonas-jespersen** [comment]
Thanks. I can confirm that it works again now in my setup.
